lib/pharos/phases/validate_version.rb

```diff
@@ -25,7 +25,7 @@ def validate_version(cluster_version)
         cluster_version = build_version(cluster_version)
         raise "Downgrade not supported" if cluster_version > pharos_version
 
-        if requirement.satisfied_by?(cluster_version)
+        if requirement.satisfied_by?(cluster_version) || cluster_version.bump == pharos_version
```

Comment:
Is the intent here to allow upgrades from `1.0.1` to `1.1.0`, but not `1.0.1` to `1.1.1`? That is what it seems to do.

this seems to do it in one conditional:

```ruby
def requirement_satisfied?
  Gem::Requirement.new('<= ' + pharos_version.to_s, ' <= ' + pharos_version.bump.to_s).satisfied_by?(existing_version)
end
```

Reply:
These should be allowed without warning (btw I added specs for these):
- `1.0.1` -> `1.0.2`
- `1.0.1` -> `1.0.3`
- `1.0.1` -> `1.1.0`

These should throw a warning:
- `1.0.1` -> `1.2.0`

Reply:
And not `1.0.99` -> `1.1.1`

Reply:
Good catch 👏 Should be fixed now.